detoks/src/main/java/nl/tudelft/trustchain/detoks/TransactionEngine.kt

```diff
@@ -0,0 +1,117 @@
+package nl.tudelft.trustchain.detoks
+
+import nl.tudelft.ipv8.Community
+import nl.tudelft.ipv8.Overlay
+import nl.tudelft.ipv8.Peer
+import nl.tudelft.ipv8.attestation.trustchain.BlockBuilder
+import nl.tudelft.ipv8.attestation.trustchain.TrustChainBlock
+import nl.tudelft.ipv8.attestation.trustchain.payload.HalfBlockBroadcastPayload
+import nl.tudelft.ipv8.attestation.trustchain.payload.HalfBlockPayload
+import nl.tudelft.ipv8.messaging.Packet
+import nl.tudelft.ipv8.util.random
+import java.util.*
+
+
+open class TransactionEngine (override val serviceId: String): Community() {
+    private val broadcastFanOut = 25
+    private val ttl = 100
+
+    object MessageId {
+        const val HALF_BLOCK = 4242
+        const val HALF_BLOCK_ENCRYPTED = 4243
+        const val HALF_BLOCK_BROADCAST = 424242
+        const val HALF_BLOCK_BROADCAST_ENCRYPTED = 424243
+    }
+
+    init {
+        messageHandlers[msgIdFixer(MessageId.HALF_BLOCK)] = ::onHalfBlockPacket
+        messageHandlers[msgIdFixer(MessageId.HALF_BLOCK_BROADCAST)] = ::onHalfBlockBroadcastPacket
+        messageHandlers[msgIdFixer(MessageId.HALF_BLOCK_ENCRYPTED)] = ::onHalfBlockPacket
+        messageHandlers[msgIdFixer(MessageId.HALF_BLOCK_BROADCAST_ENCRYPTED)] = ::onHalfBlockBroadcastPacket
+    }
+
+    fun sendTransaction(blockBuilder: BlockBuilder, peer: Peer?, encrypt: Boolean = false) {
+        println("sending block...")
```

Comment:
Debug print ? Use logger if needed

Reply:
I agree. I will use a logger from the `mu.KotlinLogging` library